scripts/list_package_versions.py

```diff
@@ -1,66 +1,167 @@
 #!/usr/bin/env python3
 
 import argparse
+import json
+import os
+import subprocess
 from pathlib import Path
+from pprint import pprint
+
 import yaml
-import json
 
+script_path = Path(__file__).resolve().parent
+root_path = script_path.parent
+recipes_path = root_path / "recipes"
 
-def get_package_infos():
-    script_path = Path(__file__).resolve().parent
-    root_path = script_path.parent
-    recipes_path = root_path / "recipes"
 
+def get_package_infos():
     package_infos = {}
 
-    for recipe_path in recipes_path.iterdir():
-        if not recipe_path.is_dir():
+    for package_path in recipes_path.iterdir():
+        if not package_path.is_dir():
             continue
 
-        infos = []
+        config_file = package_path / "config.yml"
+        if not config_file.is_file():
+            continue
 
-        with open(recipe_path / "config.yml") as f:
+        with open(config_file) as f:
             config = yaml.safe_load(f)
 
+        infos = []
+        package_name = package_path.name
         for version, details in config["versions"].items():
             infos.append(
                 {
+                    "package": package_name,
                     "version": version,
-                    "folder": Path(details["folder"]),
+                    "package_reference": "{}/{}".format(package_name, version),
+                    "conanfile": str((package_path / details["folder"] / "conanfile.py").relative_to(root_path)),
+                    "test_conanfile": str((package_path / details["folder"] / "test_package" / "conanfile.py").relative_to(root_path)),
```

Comment:
potentially this can be pushed into the build workflow with another script

Reply:
Rephrase, so I could understand better what did you mean

Reply:
So instead of resolving the path here and putting it into the job options we could have another script that determines the `test_conanfile` path from the `conanfile` path and run it in the other jobs. Nothing critical but it could reduce a bit the params passed down to other jobs and to stick with the crucial information